src/ActivitiesSDKAdapter.test.js

```diff
@@ -226,4 +214,146 @@ describe('Activities SDK Adapter', () => {
       );
     });
   });
+
+  describe('hasAdaptiveCard()', () => {
+    test('returns true if Activity object has a card attachmet', () => {
+      const hasCard = activitiesSDKAdapter.hasAdaptiveCard(mockSDKCardActivity);
+
+      expect(hasCard).toBeTruthy();
+    });
+
+    test('returns a falsy value if Activity object has not a card attachmet', () => {
+      const hasCard = activitiesSDKAdapter.hasAdaptiveCard({});
+
+      expect(hasCard).toBeFalsy();
+    });
+  });
+
+  describe('getAdaptiveCard()', () => {
+    test('returns the card object if Activity object has a attachmet of this type', () => {
+      const cardAttachment = activitiesSDKAdapter.getAdaptiveCard(mockSDKCardActivity);
+
+      expect(cardAttachment).toMatchObject({
+        $schema: 'http://adaptivecards.io/schemas/adaptive-card.json',
+        type: 'AdaptiveCard',
+        version: '1.2',
+        body: [
+          {
+            type: 'TextBlock',
+            text: 'Adaptive Cards',
+            size: 'large',
+          },
+        ],
+        actions: [
+          {
+            type: 'Action.OpenUrl',
+            url: 'http://adaptivecards.io',
+            title: 'Learn More',
+          },
+        ],
+      });
+    });
+
+    test('returns undefined if Activity object has a attachmet of card type', () => {
+      const cardAttachment = activitiesSDKAdapter.getAdaptiveCard(activityWithoutCard);
+
+      expect(cardAttachment).toBeUndefined();
+    });
+  });
+
+  describe('attachAdaptiveCard()', () => {
+    test('add an adaptive card attachment to the activity to be posted', () => {
```

Comment:
without "to be posted". attachAdaptiveCard() does not care what you do with the attachment